### ISSUE TITLE
Accessibility request: Text-to-speech

### ISSUE THREAD

**Issue by GH023** [microsoft/vscode#141213]
Dear VS Code team,

I would like to request a text-to-speech functionality with word highlighting in VS code. Such functionality will be very beneficial for people with disabilities, particularly dyslexics.
I utilize the text-to-speech functionality built-in in Windows 10 that is accessible through the 'Magnifier' feature. It has been very helpful for me (and I'm sure many others like me as well). It allows me to use a keyboard shortcut <Ctrl Alt Enter> to read anything on the desktop. It works in most of Windows programs including Microsoft Word, Notepad, WordPad, Chrome browser, etc. This is how I use it - I first left-click on a word or the beginning of a sentence where I want the engine to start reading. Then I press the hotkey. It starts to read from the focus point, utilizing the Windows inbuilt voices and highlighting the words as they are being read with a blue outline box. It has been immensely helpful.

However, this feature does not work in VS code. To be precise, it works only in specific sections of VScode. For example, I'm able to use it when reading the description of extensions. However, it does not work in the program source code files or the terminal window. So as of now, I need to manually copy the program code or terminal output to Microsoft Word and utilize the aforementioned text-to-speech functionality in there. It is a cumbersome process and slows down the workflow. Furthermore, I also lose the formatting and line numbers, as well as the context when using a third-party editor to read the text.

Use Case: By enabling this functionality in the source code as well as terminal window, I can simply click on any piece of code and read it immediately (with text to speech). 

I believe this will be a very helpful functionality for people who need text to speech while programming or debugging.

Thank you for your consideration.

**Comment by isidorn** [comment]
@GH023 thanks for creating this issue and I understand the use case and I think it makes good sense.
Let me try to summarise: the issue is that the Windows Magnifier text to speech feature does not work when focus is in the VS Code editor or terminal?
Does the text to speech of Windows Magnifier work when you go to [this site](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea) and try to read what is in the text area on the right "It was a dark and stormy night..."?

**Comment by GH023** [comment]
Hi @isidorn 
Many thanks for the quick reply and for acknowledging the issue.
I tried to use the text to speech on the site above, and it was able to read the text "It was a dark and stormy night..." along with word highlighting. There were no issues.

**Comment by GH023** [comment]
@isidorn Also, please do let me know if more information is needed at any time. Happy to help in any way I can. Hopefully, there is an easy fix that can be deployed quickly. It will significantly improve the programming experience for dyslexics.

**Comment by isidorn** [comment]
@GH023 thanks a lot for providing more information. I will have to check with the Windows Magnifier team. I will let you know what they say. I just tried to make the magnifier read anything on my Win VM and was unsuccessful.

As a workaround you can try using the [NVDA screen reader](https://www.nvaccess.org/) for reading. VS Code works best with that one on windows.

**Comment by isidorn** [comment]
@GH023 can you reproduce the issue if you set	`"editor.accessibilitySupport": "on"` in vscode settings?
